File: Medium/2026-02-12-3713-Longest-Balanced-Substring-I/solution.java
class Solution {
    public int longestBalancedSubstring(String s) {
        int maxLength = 0;
        int n = s.length();
        
        for (int len = 1; len <= n; len++) {
            for (int start = 0; start <= n - len; start++) {
                String substring = s.substring(start, start + len);
                if (isBalanced(substring)) {
                    maxLength = Math.max(maxLength, len);
                }
            }
        }
        
        return maxLength;
    }
    
    private boolean isBalanced(String substring) {
        int[] count = new int[26];
        for (char c : substring.toCharArray()) {
            count[c - 'a']++;
        }
        
        int frequency = 0;
        boolean first = true;
        
        for (int c : count) {
            if (c > 0) {
                if (first) {
                    frequency = c;
                    first = false;
                } else if (c != frequency) {
                    return false;
                }
            }
        }
        
        return true;
    }
}

File: Medium/2026-02-12-3713-Longest-Balanced-Substring-I/solution.py
class Solution:
    def longestBalancedSubstring(self, s: str) -> int:
        max_length = 0
        n = len(s)

        for length in range(1, n + 1):
            count = {}
            for i in range(n):
                if i + length <= n:
                    substring = s[i:i + length]
                    count.clear()
                    for char in substring:
                        count[char] = count.get(char, 0) + 1
                    
                    if len(set(count.values())) == 1:
                        max_length = max(max_length, length)

        return max_length